sdk/pinocchio/src/account_info.rs

```diff
@@ -609,7 +609,7 @@ impl AccountInfo {
     }
 
     /// Returns the memory address of the account data.
-    fn data_ptr(&self) -> *mut u8 {
+    pub fn data_ptr(&self) -> *mut u8 {
```

Comment:
Ok, so what about adding a "warning"?
```suggestion
    /// # Important
    ///
    /// Obtaining the raw pointer itself is safe, but dereferencing it requires
    /// the caller to uphold Rust’s aliasing rules. It is undefined behavior to dereference
    /// the pointer or write through it while any safe reference (e.g., from any of `borrow_data`
    /// or `borrow_mut_data` methods) to the same data is still alive.
    pub fn data_ptr(&self) -> *mut u8 {
```